code/modules/projectiles/projectile.dm

```diff
@@ -822,6 +833,11 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 //----------------------------------------------------------
 
 /atom/proc/bullet_act(obj/projectile/proj)
+	if(((proj.ammo.flags_ammo_behavior & AMMO_PASS_THROUGH_MOVABLE) || (proj.ammo.flags_ammo_behavior & AMMO_PASS_THROUGH_TURF)) && HAS_TRAIT(proj, TRAIT_PROJ_HIT_SOMETHING))
+		proj.damage *= proj.ammo.on_pierce_multiplier
+		proj.penetration *= proj.ammo.on_pierce_multiplier
+		proj.sundering *= proj.ammo.on_pierce_multiplier
+	ADD_TRAIT(proj, TRAIT_PROJ_HIT_SOMETHING, TRAIT_PROJ_HIT_SOMETHING)
```

Comment:
it should be ADD_TRAIT(proj, TRAIT_PROJ_HIT_SOMETHING, BULLET_ACT_TRAIT)

the last argument is the "origin" of the trait